services/org/user.go

```diff
@@ -64,10 +64,15 @@ func RemoveOrgUser(ctx context.Context, org *organization.Organization, user *us
 	if err != nil {
 		return fmt.Errorf("AccessibleReposEnv: %w", err)
 	}
-	repoIDs, err := env.RepoIDs(ctx, 1, org.NumRepos)
-	if err != nil {
-		return fmt.Errorf("GetUserRepositories [%d]: %w", user.ID, err)
+	var repoIDs []int64
+	// mssql does not support fetching 0 repositories
+	if org.NumRepos > 0 {
+		repoIDs, err = env.RepoIDs(ctx, 1, org.NumRepos)
```

Comment:
I don't think the fix is right. What if `NumRepos` is out-of-sync? Such problems happened a lot.

The root problem is not related to `org.NumRepos`, the real problem is that `RepoIDs` should have a correct behavior

Reply:
See https://github.com/go-gitea/gitea/pull/34449/commits/1c32f8cffe582933aa0db3edee94aebb7f898b9a